packages/web/src/components/search/addons/ImageDropdown.js

```diff
@@ -0,0 +1,297 @@
+			clearTimeout(urlValueTimer.current);
+		}
+		urlValueTimer.current = setTimeout(async () => {
+			if (imageURL) {
+				try {
+					const response = await fetch(imageURL);
+					const blob = await response.blob();
+					const fileName = imageURL.split('/').pop();
+					const fileType = blob.type;
+
+					const file = new File([blob], fileName, { type: fileType });
+					readFile(file);
+				} catch (e) {
+					console.error(e);
+				}
+			}
+		}, DEBOUNCE_DELAY);
+	};
+
+	const handleFileSelect = () => {
+		const fileInput = fileInputRef.current;
+		if (fileInput) {
+			const file = fileInput.files[0];
+			readFile(file);
+		}
+	};
+	const handleFileDrop = (ev) => {
+		// Prevent default behavior (Prevent file from being opened)
+		ev.preventDefault();
+		if (!imageValue) {
+			if (ev.dataTransfer.items) {
+		  // Use DataTransferItemList interface to access the file(s)
+		  [...ev.dataTransfer.items].forEach((item) => {
+					// If dropped items aren't files, reject them
+					if (item[0].kind === 'file') {
+						const file = item[0].getAsFile();
+						// TODO: Below is same as onChange
+						readFile(file);
+			  }
+		  });
+			} else {
+		  // Use DataTransfer interface to access the file(s)
+		  [...ev.dataTransfer.files].forEach((file) => {
+					readFile(file);
+		  });
+			}
+		}
+		setIsDropping(false);
+	  };
+
+	  const handleDrag = (e) => {
+		e.preventDefault();
+		if (containerRef.current.contains(e.target) && !isDropping) {
+			setIsDropping(true);
+		}
+	  };
+
+	return (
+		<Container
+			onDrop={handleFileDrop}
+			onDragOver={handleDrag}
+			ref={containerRef}
+		>
+			{
+				isDropping
+					? (
+						<DropPlaceholder >
+							Drop an image here
+						</DropPlaceholder>)
+					: (
+						<div >
+							{imageValue && !showError
+								? (
+									<Preview>
+										<PreviewImg ref={imageRef} src={imageValue} />
+										<StyledDeleteIcon onClick={handleDelete} />
+									</Preview>
+								) : null}
+
+							{!imageValue
+								? (
+									<Container.Body>
+										{showError ? <ImageError /> : <Placeholder />}
+										<PlaceholderText>
+											<span>Drag an image here or </span>
+											{/* eslint-disable-next-line jsx-a11y/label-has-for */}
+											<Label>
+												upload a file
+												<FileInput ref={fileInputRef} onChange={handleFileSelect} type="file" accept=".svg,.png,.jpg,.jpeg,.bmp,.tif,.webp" />
+											</Label>
+										</PlaceholderText>
+										<ORDivider>
+											<ORDivider.Divider />
+											<ORDivider.Text>OR</ORDivider.Text>
+											<ORDivider.Divider />
+										</ORDivider>
+										<URLInput
+											type="text"
+											placeholder="Paste an image link"
+											value={url}
+											onChange={(e) => {
+												setURL(e.target.value);
+												fetchImageFromURL(e.target.value);
+											}}
+										/>
+									</Container.Body>
+								) : null}
+						</div>)
+			}
+		</Container>
```

Comment:
@SavvyShah Please refactor this to avoid nested ternary operators, the readability is affected